### PR TITLE
rclone downloader

### LINKED ISSUE #3: Unable to Sign in and Verify for Downloading
Hello, 

I am trying to download the dataset with the SDK. However, when I try to sign in my Google account and get the verification code, the browser shows a message saying ''We have temporarily disabled Sign in with Google Account for this app''. Could you please help fix this problem?

Thank you.

### PR DESCRIPTION
**Pull request by dbarnes**
This PR changes the gdrive downloader to rclone to fix #3 (albeit perhaps temporarily)

### Notes

- This pr now uses rclone (https://github.com/rclone/rclone) for downloading
- All temporary files (rclone, patched rclone installer, rclone config) are stored in the requested download folder to be as transparent and readable to end users. This results in the following files alongside the dataset downloads:
  - ~~`patched_rclone_install.sh` - patched installed to install rclone to the download directory~~
  - `rclone` - rclone binary
  - `rclone_rrcd.conf`- rclone configuration. Intentionally separate to the global rclobe conf to avoid any unintended system changes for end users if they also use rclone. We explicitly target this configuration for downloading. Hence downloading to a new folder means re authorising rclone
  - ~~`share` - share folder from rclone~~

### Questions
- ~~Currently this script retrieves the latest version of rclone and the install script before patching in lines 99-105 in `gdrive_handler.py`. As rclone updates this script it may break our patching. Does this need to be locked down?~~
- ~~Perhaps keeping our own version of the install script pinning it to a specific rclone binary version and accept that we may have to update that separately in the future?~~
- This pr now includes a patched rclone installer (and the MIT license from their github repo) which accepts a destination install dir (where we download files) and a rclone version (we set this to the latest version v1.50.2)
- This may eventually fail when this version gets too old to interface with Google Drive but hopefully this will be a trivial edit to update to a later version as long as the API permission are still sufficient via rclone

### Install branch
```
pip install git+https://git@github.com/dbarnes/radar-robotcar-dataset-sdk.git@rclone_downloader
```
### Test some downloading
```
# Single sample
python -m radar_robotcar_dataset_sdk.downloader.download --download_folder=<DOWNLOAD_DIR> --sample_dataset=Small

# Single dataset
python -m radar_robotcar_dataset_sdk.downloader.download --download_folder=<DOWNLOAD_DIR> --datasets=2019-01-10-14-36-48-radar-oxford-10k-partial

# Two datasets
python -m radar_robotcar_dataset_sdk.downloader.download --download_folder=<DOWNLOAD_DIR> --datasets="2019-01-11-14-02-26-radar-oxford-10k,2019-01-10-14-36-48-radar-oxford-10k-partial" 

# Single sensors (radar)
python -m radar_robotcar_dataset_sdk.downloader.download --download_folder=<DOWNLOAD_DIR> --datasets=2019-01-10-11-46-21-radar-oxford-10k --sensors="Navtech CTS350-X Radar"

# Some sensors (grasshoppers)
python -m radar_robotcar_dataset_sdk.downloader.download --download_folder=<DOWNLOAD_DIR> --datasets=2019-01-10-11-46-21-radar-oxford-10k --sensors="Grasshopper 2 Left,Grasshopper 2 Right,Grasshopper 2 Rear"

# Some sensors and some datasets
python -m radar_robotcar_dataset_sdk.downloader.download --download_folder=<DOWNLOAD_DIR> --datasets="2019-01-10-14-36-48-radar-oxford-10k-partial,2019-01-10-14-36-48-radar-oxford-10k-partial" --sensors="Velodyne HDL-32E Left Pointcloud,Velodyne HDL-32E Right Pointcloud,NovAtel GPS / INS"
```
### Todo
Please could you both:
- ~~discuss patching / versioning in questions~~
- review the code changes and comments to make sure they make sense
- double check this works on osx / linux for you guys
- double check the patching of the rclone installer is sane and correct (`install.sh`)

### CODE REVIEW

**Comment by dbarnes** [comment]
@mttgdd / tom please do not review yet. After discussion I am patching to a fixed install script and rclone version.

**Comment by dbarnes** [comment]
The PR has now been updated to used a fixed rclone version (v1.50.2 which is the current latest) and is ready for review.

**Comment by mttgdd** [comment]
I have tested this with `--sample_dataset Sample` and it works fine.

However, with the following flags:

```bash
--datasets="2019-01-16-11-53-11-radar-oxford-10k"
--sensors="Navtech CTS350-X Radar"
```

I get this error message:

```bash
====================================================================================================
Initialising rclone to: /Users/mattgadd/Downloads/oxford-radar-robotcar-dataset
====================================================================================================
rclone install script: installed version of rclone is already at requested version: v1.50.1

====================================================================================================
rclone binary at : /Users/mattgadd/Downloads/oxford-radar-robotcar-dataset/rclone
====================================================================================================

====================================================================================================
rclone config at: /Users/mattgadd/Downloads/oxford-radar-robotcar-dataset/rclone_rrcd.conf
====================================================================================================

Downloading    0 /    1 : 2019-01-16-11-53-11-radar-oxford-10k             - Navtech CTS350-X Radar    - 3.3 GB            - https://drive.google.com/file/d/1bQviDEkU20uq-xGeHrciaA5YeOr1-uyW

2019-12-12 14:48:06 ERROR : : error reading source directory: directory not found
2019-12-12 14:48:06 ERROR : Attempt 1/3 failed with 2 errors and: directory not found
2019-12-12 14:48:06 ERROR : : error reading source directory: directory not found
2019-12-12 14:48:06 ERROR : Attempt 2/3 failed with 2 errors and: directory not found
2019-12-12 14:48:06 ERROR : : error reading source directory: directory not found
2019-12-12 14:48:06 ERROR : Attempt 3/3 failed with 2 errors and: directory not found
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Errors:                 2 (retrying may help)
Checks:                 0 / 0, -
Transferred:            0 / 0, -
Elapsed time:          0s
2019/12/12 14:48:06 Failed to copy with 2 errors: last error was: directory not found
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/mattgadd/code/github/dbarnes/radar-robotcar-dataset-sdk/radar_robotcar_dataset_sdk/downloader/download.py", line 193, in <module>
    app.run(main)
  File "/usr/local/lib/python3.6/site-packages/absl/app.py", line 300, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.6/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/Users/mattgadd/code/github/dbarnes/radar-robotcar-dataset-sdk/radar_robotcar_dataset_sdk/downloader/download.py", line 180, in main
    downloaded_zip_file_path = gdrive_handler.download_filename(filename)
  File "/Users/mattgadd/code/github/dbarnes/radar-robotcar-dataset-sdk/radar_robotcar_dataset_sdk/downloader/gdrive_handler.py", line 46, in download_filename
    subprocess.check_call(args)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/Users/mattgadd/Downloads/oxford-radar-robotcar-dataset/rclone', '--config', '/Users/mattgadd/Downloads/oxford-radar-robotcar-dataset/rclone_rrcd.conf', '--drive-shared-with-me', '--progress', 'copy', 'rrcd_drive:2019-01-16-11-53-11-radar-oxford-10k_Navtech_CTS350-X_Radar', '/Users/mattgadd/Downloads/oxford-radar-robotcar-dataset']' returned non-zero exit status 3.
```

**Comment by mttgdd** [comment]
I was using the wrong email address (one I did not register for the dataset with).

I have now tested:

```bash
python -m radar_robotcar_dataset_sdk.downloader.download --datasets="2019-01-16-11-53-11-radar-oxford-10k" --sensors="Navtech CTS350-X Radar" --download_folder ~/Downloads/oxford-radar-robotcar-dataset
```

And the radar scans were downloaded successfully.

File: README.md [hunk 1]
```diff
@@ -9,7 +9,12 @@ Matlab and Python parsers for all the data are included in the original [**Oxfor
 
 ** Please Read **
 --------------
-The existing downloader relies on an external binary for downloading the dataset which is currently unable to authorise users as described on this [issue](https://github.com/dbarnes/radar-robotcar-dataset-sdk/issues/3). We have written our own downloader but we must wait to be verified by Google for this to be used externally. The meantime please download files manually from the dataset website.
+Although the automatic downloader script and other code has been tested extensively we accept no liability for use of the code.
+If you have any concern please review the source code before using.
```
Comment (mttgdd) [inline README.md]: concerns

File: README.md [hunk 2]
```diff
@@ -9,7 +9,12 @@ Matlab and Python parsers for all the data are included in the original [**Oxfor
...
+
+Using [**rclone**](https://github.com/rclone/rclone) (the binary used to download the dataset files) we target download files by filename rather than id.
+If you have other files shared with you in Google Drive that match our dataset filenames you may encounter unintended errors.
+However as we use long filenames such as `2019-01-15-14-24-38-radar-oxford-10k_Bumblebee_XB3_Visual_Odometry` we believe this is unliekely and additionally we use read only permission for additional safety.
```
Comment (mttgdd) [inline README.md]: unlikely

File: README.md [hunk 3]
```diff
@@ -92,26 +98,37 @@ python -m radar_robotcar_dataset_sdk.downloader.download \
...
 
-We provide authorised downloads via Google Drive for each registered user. To automatically handle this we use a forked version of GDrive with Team Drive support which can be found at: `https://github.com/petrpulc/gdrive`. This binary is automatically downloaded and run as part of the download script. To use this binary you will need to give permission for the binary to use your credentials. When prompted simply follow the instructions (as shown below) by:
+We provide authorised downloads via Google Drive for each registered user.
+To automatically handle this we use rclone which can be found at: `https://github.com/rclone/rclone`.
+This binary is automatically downloaded and run as part of the download script.
+To use this binary you will need to give permission for the binary to use your credentials.
+When prompted simply follow the instructions (as shown below) by:
 
-- Going to the url in your browser
+- Going to the url in your browser (opened automatically locally in a browser if run locally)
```
Comment (mttgdd) [inline README.md]:
```suggestion
Going to the url in your browser (opened automatically in a browser if run locally)
```

**Review by mttgdd — APPROVED** [review]

**Pull request merged**